octodns/provider/mythicbeasts.py

```diff
@@ -0,0 +1,420 @@
+#
+#
+#
+
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+import re
+
+from requests import Session
+from logging import getLogger
+
+from ..record import Record
+from .base import BaseProvider
+
+
+def add_trailing_dot(value):
+    '''
+    Add trailing dots to values
+    '''
+    assert value, 'Missing value'
+    assert value[-1] != '.', 'Value already has trailing dot'
+    return value + '.'
+
+
+def remove_trailing_dot(value):
+    '''
+    Remove trailing dots from values
+    '''
+    assert value, 'Missing value'
+    assert value[-1] == '.', 'Value already missing trailing dot'
+    return value[:-1]
+
+
+class MythicBeastsProvider(BaseProvider):
+    '''
+    Mythic Beasts DNS API Provider
+
+    mythicbeasts:
+      class: octodns.provider.mythicbeasts.MythicBeastsProvider
+        zones:
+          my-zone: 'password'
+    '''
+
+    SUPPORTS_GEO = False
+    SUPPORTS_DYNAMIC = False
+    SUPPORTS = set(('A', 'AAAA', 'ALIAS', 'CNAME', 'MX', 'NS',
+                    'SRV', 'SSHFP', 'CAA', 'TXT'))
+    BASE = 'https://dnsapi.mythic-beasts.com/'
+
+    def __init__(self, identifier, passwords, *args, **kwargs):
+        self.log = getLogger('MythicBeastsProvider[{}]'.format(identifier))
+
+        assert isinstance(passwords, dict), 'Missing passwords'
```

Comment:
Probably best to be a little more specific in what's happening here. If someone did:

```yaml
  passwords: foobarbaz
```

this would tell them that they're missing passwords when the problem is really that passwords must be a dictionary. The `passwords` param up above should serve as the "requires" part. This should probably tell them that passwords must be a dict.

```suggestion
        assert isinstance(passwords, dict), 'passwords must be a dictionary'
```

Reply:
Will do, and update tests